### PR TITLE
[DinoMod] dino weak point proficiencies

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary
Mods "[DinoMod] dino weak point proficiencies"

#### Purpose of change

Feature parity with main game

#### Describe the solution

Takes existing weak point proficiencies from #56487 and applies them to dinos and zinos. Creates new dino proficiency and assigns it to living dinos and juveniles. Reverts the looks like for apatosaurus to fix sauropod art in UDP

#### Describe alternatives you've considered

N/A

#### Testing

Game loads no errors, monsters spawn and look right

#### Additional context

Spent way too much time trying to find a funny quote here.

### CODE REVIEW

**Comment by dseguin** [comment]
Looks good! The only thing I would say is that `prof_intro_biology` doesn't give any bonuses for hitting weakpoints (it's only used as a requirement for other weakpoint proficiencies). What I would do is add a new proficiency for Dino Biology, something like:

```JSON
{
  "type": "proficiency",
  "id": "prof_wp_dino",
  "category": "prof_weakpoint",
  "name": { "str": "Dino Biology" },
  "description": "You're familiar with general dinosaur anatomy, and can exploit potential weaknesses.",
  "required_proficiencies": [ "prof_intro_biology" ],
  "can_learn": true,
  "time_to_learn": "6 h",
  "default_weakpoint_bonus": 4,
  "default_weakpoint_penalty": -2
}
```

And then add it to the dino "families" fields:

```JSON
"families": [ "prof_intro_biology", "prof_wp_dino" ]
```

**Comment by LyleSY** [comment]
Done, thanks!

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 1]
```diff
@@ -891,7 +909,6 @@
     "type": "MONSTER",
     "id": "mon_apatosaurus",
     "name": { "str_sp": "apatosaurus" },
-    "looks_like": "mon_camptosaurus",
```
Comment (dseguin) [inline data/mods/DinoMod/monsters/dinosaur.json:894]: Just checking, is this intentional?

Comment (LyleSY) [reply data/mods/DinoMod/monsters/dinosaur.json:894]: Yep, it was on the list waiting for the next PR, but this became the next PR

**Review by dseguin — APPROVED** [review]
Cheers, LGTM

**Comment by LyleSY** [comment]
Build failures don't look related

**Pull request merged**